src/Command/UsersCheckPasswordCommand.php

```diff
@@ -101,7 +101,7 @@ protected function execute(InputInterface $input, OutputInterface $output): ?int
 
         // Check if user exists
         $user = $this->repository->findByEmail($email);
```

Comment:
I would prefer to adjust `findByEmail` to take an optional argument `deleted=false`.
Filtering this out afterwards looks potentially unsafe (as we can see in this PR) and counter-intuitive.

Reply:
I also don't like the different handling of user authentication. We currently have two implementations: `UsersCheckPasswordCommand` with a self-built solution and the Form Login from Symfony. This uses the Framework Security Chain for authentication. I would rather use this mechanism here in the `UsersCheckPasswordCommand` instead of adjusting the `UserRepository`. It seems to me that it makes more sense to use Symfony's internal logic instead of building a solution yourself. We will stumble over this again and again.

WDYT?